### ISSUE TITLE
[Spring] generate paginated resources

### ISSUE THREAD

**Issue by cbornet** [swagger-api/swagger-codegen#3353]
To follow the discussion from #3133 
The idea is to generate Page/Pageable if the operation has a "x-spring-paginated" vendor-extension.

I just tried
```java
    public ResponseEntity<List<Pet>> findPetsByTags(@ApiParam(value = "Tags to filter by", required = true) @RequestParam(value = "tags", required = true) List<String> tags, Pageable pageable) {
        ArrayList<Pet> petList = new ArrayList<Pet>();
        Page<Pet> page = new PageImpl(petList, pageable, 0);
        // do some magic!
        return new ResponseEntity<List<Pet>>(page.getContent(), HttpStatus.OK);
    }
```
in spring-boot after including spring-data-commons and didn't have any issue.
I didn't have to configure any datasource.

**Comment by cbornet** [comment]
@diyfr

**Comment by diyfr** [comment]
Generate code works [https://github.com/diyfr/swagger-codegen/commit/e908589f2b8cda673e3c85c42ac309916ab41a6f](first commit x-spring-pageable).
What dependency used for your test ?
I used spring-boot-starter-data-jpa , perhaps this is the problem?

**Comment by diyfr** [comment]
[https://github.com/swagger-api/swagger-codegen/pull/3357](https://github.com/swagger-api/swagger-codegen/pull/3357)

**Comment by cbornet** [comment]
@diyfr Don't use spring-boot-starter-data-jpa, use only spring-data-commons

**Comment by cbornet** [comment]
As for naming I prefer "x-spring-paginated" than "x-spring-pageable"

**Comment by cbornet** [comment]
Also I tested the the plugin for Pageable parameters and it worked great. I find it cleaner than @ApiImplicitParam everywhere in the code.

**Comment by diyfr** [comment]
@cbornet i use x-spring-pageable because is the know spring component.
And plugin need to rebuild springfox or can it in springboot project ?

**Comment by cbornet** [comment]
No you can just include the class in your project and a @Bean in a @Configuration to activate it.

**Comment by cbornet** [comment]
Yes Pageable is the name of a component used to implement pagination (you also need Page) and it doesn't describe the nature of the operation which is that it is "paginated"

**Comment by diyfr** [comment]
@cbornet ok for x-spring-paginated.
I do not know if I'll have time for the plug-in today
